static-analysis/autodispose-lint/src/main/kotlin/com/uber/autodispose/lint/AutoDisposeDetector.kt

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (C) 2018. Uber Technologies
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.uber.autodispose.lint
+
+import com.android.tools.lint.client.api.JavaEvaluator
+import com.android.tools.lint.detector.api.*
+import com.intellij.psi.PsiClass
+import com.intellij.psi.PsiMethod
+import org.jetbrains.uast.UCallExpression
+import org.jetbrains.uast.getContainingClass
+import org.jetbrains.uast.getContainingUClass
+import java.lang.IllegalStateException
+import java.util.EnumSet
+
+/**
+ * Detector which checks if your stream subscriptions are handled by AutoDispose.
+ */
+class AutoDisposeDetector: Detector(), SourceCodeScanner {
+
+  companion object {
+    val ISSUE: Issue = Issue.create(
+        "AutoDisposeUsage",
+        "Always apply an AutoDispose scope before subscribing within defined scoped elements.",
```

Comment:
Typically this should be shorter since it shows up in the Android Studio IDE

Reply:
Hmmm it doesn't seem too long in the screenshot I attached in the thread above. Basically copied the message from the error-prone check. @hzsweers Any preference?

Reply:
Let's keep it as is, maybe with a linebreak after the first sentence? The IDE does the thing where it says you can read more right?

Reply:
It's only one sentence though. The way it works is `Always apply an AutoDispose scope before subscribing within defined scoped elements.` comes as the primary hint. When you click "Read More", you're given the detailed message in line 41. `You're subscribing to an observable but not handling it's subscription. This can result in memory leaks. You can avoid memory leaks by appending .as(autoDisposable(this)) before you subscribe.`